### PR TITLE
ECO-2522: added neccessary data to request to make it valid for easycredit

### PR DESCRIPTION
**Pull request by Ruslan05**
- Developer(s): @Ruslan05

- Ticket: https://spryker.atlassian.net/browse/ECO-2521

- Academy PR: ACADEMY_URL_HERE

- Release Group: URL_HERE

#### Please confirm

- [ ] No new OS components - or they have been approved by legal department

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   Computop              | minor                 |                       |

#### Release Notes

{skip}

-----------------------------------------

##### Change log

Improvements

- Added `shippingCity`, `shippingStreetNumber `, `shippingStreet `  and `shippingCountryCode ` properties to `ComputopEasyCreditPayment `.
- Adjusted `SprykerEco\Yves\Computop\Mapper\Init\PostPlace\ EasyCreditMapper ` to send additional data mentioned in transfer.

### CODE REVIEW

File: src/SprykerEco/Yves/Computop/Mapper/Init/PostPlace/EasyCreditMapper.php [hunk 1]
```diff
@@ -90,7 +93,47 @@ protected function getDataSubArray(ComputopEasyCreditPaymentTransfer $cardPaymen
...
+     * @param \Generated\Shared\Transfer\ComputopEasyCreditPaymentTransfer $computopEasyCreditPaymentTransfer
+     *
+     * @return \Generated\Shared\Transfer\ComputopEasyCreditPaymentTransfer
+     */
+    protected function mapAddressDataToEasyCreditPayment(
+        array $addressData,
+        ComputopEasyCreditPaymentTransfer $computopEasyCreditPaymentTransfer
+    ): ComputopEasyCreditPaymentTransfer {
+
+        $computopEasyCreditPaymentTransfer->setShippingCity($addressData[ComputopEasyCreditPaymentTransfer::SHIPPING_CITY]);
```
Comment (gechetspr) [inline src/SprykerEco/Yves/Computop/Mapper/Init/PostPlace/EasyCreditMapper.php]: mb smth like `$addressData[ComputopEasyCreditPaymentTransfer::SHIPPING_CITY] ?? null`?

Comment (Ruslan05) [reply src/SprykerEco/Yves/Computop/Mapper/Init/PostPlace/EasyCreditMapper.php]:
```
    protected function getAdditionalAddressData(QuoteTransfer $quoteTransfer): array
    {
        $addressTransfer = $quoteTransfer->getBillingSameAsShipping() ? $quoteTransfer->getBillingAddress() : $quoteTransfer->getShippingAddress();

        $addressData[ComputopEasyCreditPaymentTransfer::SHIPPING_CITY] = $addressTransfer->getCity();
        $addressData[ComputopEasyCreditPaymentTransfer::SHIPPING_STREET] = $addressTransfer->getAddress1();
        $addressData[ComputopEasyCreditPaymentTransfer::SHIPPING_STREET_NUMBER] = $addressTransfer->getAddress2();
        $addressData[ComputopEasyCreditPaymentTransfer::SHIPPING_COUNTRY_CODE] = $addressTransfer->getIso2Code();
}
```
```
$addressData = $this->getAdditionalAddressData($quoteTransfer);
$computopPaymentTransfer = $this->mapAddressDataToEasyCreditPayment($addressData, $computopPaymentTransfer);
```

So this keys will always exist. I think it`s not needed to use ternar operation in this case. What do you think ?

Comment (aleksey-kotsuba) [reply src/SprykerEco/Yves/Computop/Mapper/Init/PostPlace/EasyCreditMapper.php]: What about using `$transfer->fromArray($array);` ?

Comment (Ruslan05) [reply src/SprykerEco/Yves/Computop/Mapper/Init/PostPlace/EasyCreditMapper.php]: There is already related property in `$computopEasyCreditPaymentTransfer` e.g. `shippingZip`.
So i`m following same structure for other adderrs data, `shippingCity` , `shippingStreet` ..

`$transfer->fromArray($array);` will not work this way due to different proprites naming: `city`, `address1`.

cc: @aleksey-kotsuba

**Pull request merged**